src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -4107,17 +4107,24 @@ extern "C" CLR_BOOL QCALLTYPE SfiNext(StackFrameIterator* pThis, uint* uExCollid
                     }
                 }
             }
+        }
 
-            if (pThis->GetFrameState() != StackFrameIterator::SFITER_FRAMELESS_METHOD)
-            {
-                retVal = pThis->Next();
-            }
+        if (pThis->GetFrameState() == StackFrameIterator::SFITER_FRAMELESS_METHOD)
```

Comment:
This is just a cosmetic change, right?

Reply:
You are probably right, I'll need to think about it for a moment. I tried to stay true to the original loop logic which may not necessarily be the optimal one.

Reply:
The original loop would call the Next and then do the rest of the code in the loop only if !pThis->m_crawl.IsFrameless(). So if it reached frameless frame, it would end up right away at the while which would exit the loop.

Reply:
Just to make sure I understand it...

If you are in the `if (!pThis->m_crawl.IsFrameless())` branch, is it necessary to check again `if (pThis->GetFrameState() != StackFrameIterator::SFITER_FRAMELESS_METHOD)` before doing the `retVal = pThis->Next();`? Are there any other states that can happen? If not, we can probably do your version and drop the extra check (+ `EH_LOG` on `SWA_FAILED`).

Reply:
Actually, if we added a break after the `_ASSERTE_MSG(pThis->GetFrameState() == StackFrameIterator::SFITER_FRAMELESS_METHOD, "Collided unwind should have reached a frameless method");`, then the check would not be needed. Without the break, it ensures we don't move one extra frame after the collided state restoration.

Reply:
Yeah, I missed that branch. That makes sense.

Reply:
How about the current version? I rewrote my commits starting from your version...